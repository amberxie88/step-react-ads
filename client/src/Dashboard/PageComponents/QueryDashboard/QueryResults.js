import React from 'react';
import Title from '../../Utilities/Title';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TablePagination from '@material-ui/core/TablePagination';

export default function QueryResults(props) {
  const { rows, fields } = props;
  const [rowsPerPage, setRowsPerPage] = React.useState(10);
  const [page, setPage] = React.useState(0);

  const handleChangePage = (event, newPage) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event) => {
    // If the selection is [A]ll the rows.
    if (event.target.value.toString().charAt(0) === 'A') {
      event.target.value = rows.length;
    }
    setRowsPerPage(+event.target.value);
    setPage(0);
  };
  return (
    <React.Fragment>
      <Title>Query Results</Title>
      <Results
        rows={rows}
        fields={fields}
        rowsPerPage={rowsPerPage}
        page={page}
        handleChangePage={handleChangePage}
        handleChangeRowsPerPage={handleChangeRowsPerPage}
      />
    </React.Fragment>
  );
}

function Results(props) {
  const rows = props.rows;
  const fields = props.fields;
  const rowsPerPage = props.rowsPerPage;
  const page = props.page;
  const handleChangePage = props.handleChangePage;
  const handleChangeRowsPerPage = props.handleChangeRowsPerPage;
  //console.log(rows);
  //console.log(fields);
  return (
    <div>
      <Table size="small">
        <TableHead>
          <TableRow>
            {fields.map((key, index) => (
              <TableCell key={'col' + index}>{key}</TableCell>
            ))}
          </TableRow>
        </TableHead>
        <TableBody>
          {rows
            .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
            .map((row) => (
              <TableRow key={'row' + row.id}>
                {fields.map((key) => (
                  <TableCell key={key}>{row[key]}</TableCell>
                ))}
              </TableRow>
            ))}
        </TableBody>
      </Table>
      <TablePagination
        rowsPerPageOptions={[
          5,
          10,
          25,
          'All ' + rows.length.toString() + ' Rows',
        ]}
        component="div"
        count={rows.length}
        rowsPerPage={rowsPerPage}
        page={page}
        onChangePage={handleChangePage}
        onChangeRowsPerPage={handleChangeRowsPerPage}
      />
    </div>
  );
}
