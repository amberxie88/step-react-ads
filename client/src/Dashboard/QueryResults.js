import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Title from './Title';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Query from './Queries';

const useStyles = makeStyles((theme) => ({
  root: {
    margin: theme.spacing(1),
  },
}));

// const rows = [];

function createData(id, result) {
  return { id, result };
}

function parseJSON(response) {
  return response.json();
}

class QueryResults extends React.Component {
  constructor(props) {
    super(props);
    this.handleQuery = this.handleQuery.bind(this);
    this.state = { rows: [] };
  }

  handleQuery() {
    alert('The button was clicked.');
    fetch('/campaign', {
      accept: 'application/json',
    })
      .then(parseJSON)
      .then((result) => {
        this.setState({
          rows: [createData(0, result)],
        });
      });
  }

  render() {
    const rows = this.state.rows;

    let button = <SubmitButton onClick={this.handleQuery} />;

    return (
      <React.Fragment>
        <Title>Query Here</Title>
        <Query />
        {button}
        <DisplayRows rows={rows} />
      </React.Fragment>
    );
  }
}

function Results(props) {
  const rows = props.rows;
  return (
    <Table size="small">
      <TableHead>
        <TableRow>
          <TableCell>Results</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {rows.map((row) => (
          <TableRow key={row.id}>
            <TableCell>{row.result}</TableCell>
          </TableRow>
        ))}
      </TableBody>
    </Table>
  );
}

function DisplayRows(props) {
  const rows = props.rows;
  return <Results rows={rows} />;
}

function SubmitButton(props) {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Button variant="outlined" onClick={props.onClick}>
        Submit
      </Button>
    </div>
  );
}

export default QueryResults;
