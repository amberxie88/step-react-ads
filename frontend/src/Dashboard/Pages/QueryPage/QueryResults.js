/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import Title from '../../Utilities/Title';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TablePagination from '@material-ui/core/TablePagination';

export default function QueryResults(props) {
  const { rows, fields } = props;

  return (
    <React.Fragment>
      <Title>Query Results</Title>
      <Results rows={rows} fields={fields} />
    </React.Fragment>
  );
}

function Results(props) {
  const { rows, fields } = props;
  const [rowsPerPage, setRowsPerPage] = React.useState(10);
  const [page, setPage] = React.useState(0);

  const handleChangePage = (event, newPage) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event) => {
    // If the selection is [A]ll the rows.
    if (event.target.value.toString().charAt(0) === 'A') {
      event.target.value = rows.length;
    }
    setRowsPerPage(+event.target.value);
    setPage(0);
  };

  const rowSlice = rows.slice(
    page * rowsPerPage,
    page * rowsPerPage + rowsPerPage,
  );
  return (
    <div>
      <Table size="small">
        <TableHead>
          <TableRow>
            {fields.map((key, index) => (
              <TableCell key={'col' + index}>{key}</TableCell>
            ))}
          </TableRow>
        </TableHead>
        <TableBody>
          {rowSlice.map((row) => (
            <TableRow key={'row' + row.id}>
              {fields.map((key) => (
                <TableCell key={key}>{row[key]}</TableCell>
              ))}
            </TableRow>
          ))}
        </TableBody>
      </Table>
      <TablePagination
        rowsPerPageOptions={[
          5,
          10,
          25,
          'All ' + rows.length.toString() + ' Rows',
        ]}
        component="div"
        count={rows.length}
        rowsPerPage={rowsPerPage}
        page={page}
        onChangePage={handleChangePage}
        onChangeRowsPerPage={handleChangeRowsPerPage}
      />
    </div>
  );
}
