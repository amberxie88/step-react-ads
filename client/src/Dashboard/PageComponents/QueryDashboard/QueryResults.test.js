import React from 'react';
import QueryResults from './QueryResults';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('QueryResults Unit Testing', () => {
  it('QueryResults properly displays table given correct input', async () => {
    const testRows = [
      { 'campaign.name': 'Sales-Search-1', 'metrics.clicks': '0', id: 0 },
    ];
    const testFields = ['campaign.name', 'metrics.clicks'];
    const component = shallow(
      <QueryResults rows={testRows} fields={testFields} />,
    );
    const componentHTML = component.html();
    expect(componentHTML).toMatchSnapshot(); //displayed state also needs to match expected results
  });
});
