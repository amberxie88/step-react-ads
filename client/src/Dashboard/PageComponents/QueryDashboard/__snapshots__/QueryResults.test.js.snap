// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryResults Unit Testing QueryResults properly displays table given correct input 1`] = `"<h2 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom\\">Query Results</h2><table class=\\"MuiTable-root\\"><thead class=\\"MuiTableHead-root\\"><tr class=\\"MuiTableRow-root MuiTableRow-head\\"><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall\\" scope=\\"col\\">campaign.name</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall\\" scope=\\"col\\">metrics.clicks</th></tr></thead><tbody class=\\"MuiTableBody-root\\"><tr class=\\"MuiTableRow-root\\"><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall\\">Sales-Search-1</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall\\">0</td></tr></tbody></table>"`;
